File: apps/desktop/src/mappers/system.ts
import type { SystemStatusResponse } from '@nova/api-contracts'
import type { SystemStatusViewModel } from '@/view-models'

export function mapSystemStatus(response: SystemStatusResponse): SystemStatusViewModel {
  return {
    status: response.status,
    version: response.version,
    subsystems: response.subsystems.map((subsystem) => ({
      name: subsystem.name,
      status: subsystem.status,
      message: subsystem.message,
    })),
  }
}

File: apps/backend/services/api/schemas/system.py
from typing import Literal, Optional

from pydantic import BaseModel, Field


SubsystemStatus = Literal["ok", "degraded", "unavailable"]


class SubsystemDiagnostic(BaseModel):
    name: str
    status: SubsystemStatus
    message: Optional[str] = None


class SystemStatusResponse(BaseModel):
    status: SubsystemStatus
    version: str
    subsystems: list[SubsystemDiagnostic] = Field(default_factory=list)
